src/category_theory/monoidal/preadditive.lean

```diff
@@ -47,4 +47,143 @@ local attribute [simp] monoidal_preadditive.tensor_add monoidal_preadditive.add_
 instance tensoring_left_additive (X : C) : ((tensoring_left C).obj X).additive := {}
 instance tensoring_right_additive (X : C) : ((tensoring_right C).obj X).additive := {}
 
+open_locale big_operators
+
+lemma tensor_sum {P Q R S : C} {J : Type*} (s : finset J) (f : P ⟶ Q) (g : J → (R ⟶ S)) :
+  f ⊗ ∑ j in s, g j = ∑ j in s, f ⊗ g j :=
+begin
+  rw ←tensor_id_comp_id_tensor,
+  let tQ := (((tensoring_left C).obj Q).map_add_hom : (R ⟶ S) →+ _),
+  change _ ≫ tQ _ = _,
+  rw [tQ.map_sum, preadditive.comp_sum],
+  dsimp [tQ],
+  simp only [tensor_id_comp_id_tensor],
+end
+
+lemma sum_tensor {P Q R S : C} {J : Type*} (s : finset J) (f : P ⟶ Q) (g : J → (R ⟶ S)) :
+  (∑ j in s, g j) ⊗ f = ∑ j in s, g j ⊗ f :=
+begin
+  rw ←tensor_id_comp_id_tensor,
+  let tQ := (((tensoring_right C).obj P).map_add_hom : (R ⟶ S) →+ _),
+  change tQ _ ≫ _ = _,
+  rw [tQ.map_sum, preadditive.sum_comp],
+  dsimp [tQ],
+  simp only [tensor_id_comp_id_tensor],
+end
+
+variables {C} [has_finite_biproducts C]
+
+/-- The isomorphism showing how tensor product on the left distributes over direct sums. -/
```

Comment:
Isn't this a special case of `⊗` preserving colimits? Which should follow quite directly from it being left adjoint... Or is the point that this definition has much better defeqs?

Reply:
Tensoring with a fixed object is not always a left adjoint! This is the nontrivial condition of being closed monoidal. 

It may be, however, that there is something more to say here: i.e. that if the category is closed monoidal, then the colimit comparison morphism for a direct sum is this isomorphism. I don't think it needs to be in this PR, however.

Reply:
Aah, of course. I was somehow only thinking of the closed monoidal case :facepalm: 

But yes, I guess in a future PR we'll want to show that this is the same as whatever rolls out of the general colimit preservation when `C` is closed monoidal.